pkg/internal/checkup/vmi/vmi.go

```diff
@@ -0,0 +1,289 @@
+/*
+ * This file is part of the kiagnose project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package vmi
+
+import (
+	corev1 "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	kvcorev1 "kubevirt.io/api/core/v1"
+)
+
+// Based on annotation names from:
+// https://github.com/cri-o/cri-o/blob/fa0fa5de1c17ddd7b6fcdbc030b6b571ce37e643/pkg/annotations/annotations.go
```

Comment:
So this solution is CRI-O specific?

Reply:
AFAIK, the VMI manifest used here was tested on OpenShift Virtualization, which uses CRI-O.

I don't know if other container runtimes have similar knobs.

Reply:
Is there a reference for using these annotations on Kubevirt docs?

Is there an API on Kubevirt to configure the VM similar to placing these annotations?

Reply:
There is this blog post about realtime VMs in KubeVirt:
https://kubevirt.io/2021/Running-Realtime-Workloads.html

It doesn't contain the cri-o annotations.

I also haven't found documentaion about these annotations in the [user guide](https://kubevirt.io/user-guide).